pdfminer/pdfdocument.py

```diff
@@ -500,28 +500,74 @@ def get_cfm(self, name):
 
     def authenticate(self, password):
         password = password.encode('utf-8')[:127]
-        hash = sha256(password)
-        hash.update(self.o_validation_salt)
-        hash.update(self.u)
-        if hash.digest() == self.o_hash:
-            hash = sha256(password)
-            hash.update(self.o_key_salt)
-            hash.update(self.u)
-            cipher = Cipher(algorithms.AES(hash.digest()),
+        hash = self._password_hash(password, self.o_validation_salt, self.u)
+        if hash == self.o_hash:
+            hash = self._password_hash(password, self.o_key_salt, self.u)
+            cipher = Cipher(algorithms.AES(hash),
                             modes.CBC(b'\0' * 16),
                             backend=default_backend())
             return cipher.decryptor().update(self.oe)
-        hash = sha256(password)
-        hash.update(self.u_validation_salt)
-        if hash.digest() == self.u_hash:
-            hash = sha256(password)
-            hash.update(self.u_key_salt)
-            cipher = Cipher(algorithms.AES(hash.digest()),
+        hash = self._password_hash(password, self.u_validation_salt)
+        if hash == self.u_hash:
+            hash = self._password_hash(password, self.u_key_salt)
+            cipher = Cipher(algorithms.AES(hash),
                             modes.CBC(b'\0' * 16),
                             backend=default_backend())
             return cipher.decryptor().update(self.ue)
         return None
 
+    def _password_hash(self, password, salt, vector=b""):
```

Comment:
Perfectly named method! Very clear :) 

I prefer to use `None` as a value to signal that it should be ignored. I also prefer explicitly checking if `vector is None`. Can you change that?

Reply:
I wanted to fix this myself but I don't have permissions to push code to your branch

Reply:
Done :-)